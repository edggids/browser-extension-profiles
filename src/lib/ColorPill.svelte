<script lang="ts">
import {createEventDispatcher} from "svelte";
import type {ColorModel} from "../models/ColorModel";

export let color:ColorModel;
export let label:string|null = null;

let className = '';
export { className as class };

const dispatch = createEventDispatcher();

$:wrapperStyle = `--color-current-light: ${color.light}`;
$:wrapperClass = `inline-flex items-center p-2 hover:bg-current-light ${className}`;

function onClick() { dispatch('click'); }
</script>

<span role="button" on:click={onClick} class={wrapperClass} style={wrapperStyle}>
    <span class="inline-flex mr-3">
        <span style="width: 16px; height:24px; background-color: {color.light}"></span>
        <span style="width: 16px; height:24px; background-color: {color.default}"></span>
        <span style="width: 16px; height:24px; background-color: {color.darker}"></span>
    </span>

    {#if label}
        <span class="text leading-none pr-3">{label}</span>
    {/if}
</span>
