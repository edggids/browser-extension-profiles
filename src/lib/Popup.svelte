<script lang="ts">
    export let isOpen = false;
    let className = '';

    export { className as class};
</script>

<div role="dialog" class={`${className} ${isOpen ? '': 'hidden'} p-2 bg-ground rounded shadow z-10`}>
    <slot></slot>
</div>

