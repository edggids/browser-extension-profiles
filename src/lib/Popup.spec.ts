import {cleanup, render} from '@testing-library/svelte';
import {describe, it, expect, afterEach} from 'vitest';
import Popup from "./Popup.svelte";
import ColorPill from "./ColorPill.svelte";

describe('Popup component', () => {
  afterEach(() => {
    cleanup();
  });

  it('mounts', () => {
    const { container, getByRole } = render(Popup, { props: { isOpen: true } });

    expect(container).toMatchSnapshot();
  });

  it('renders when isOpen is true', () => {
    const { container, getByRole } = render(Popup, { props: { isOpen: true } });
    expect(getByRole('dialog')).not.toBeNull();
  });

  it('does not render when isOpen is false', () => {
    const { container, getByRole } = render(Popup, { props: { isOpen: false } });
    expect(getByRole('dialog').classList.contains('hidden')).toBe(true);
  });

  it('applies custom class when provided', () => {
    const { container, getByRole } = render(Popup, { props: { isOpen: true, class: 'custom-class' } });
    expect(getByRole('dialog').classList.contains('custom-class')).toBe(true);
  });
});
