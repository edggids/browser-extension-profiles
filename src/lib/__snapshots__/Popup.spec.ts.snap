// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Popup component > mounts 1`] = `
<body>
  <div>
    <div
      class="  p-2 bg-ground rounded shadow z-10"
      role="dialog"
    />
  </div>
</body>
`;
